packages/loopring_v3/contracts/impl/libexchange/ExchangeDeposits.sol

```diff
@@ -118,6 +118,10 @@ library ExchangeDeposits
             feeETH
         );
 
+        // Make sure the public key can be stored in the SNARK field
+        require(account.pubKeyX < ExchangeData.SNARK_SCALAR_FIELD(), "INVALID_PUBKEY");
```

Comment:
since `ExchangeData.SNARK_SCALAR_FIELD()` is only used here, should we just declare a constant var locally?

Reply:
Should we check pubKeyX/Y before account creation and password reset instead of at places where these values are used?

Reply:
> since `ExchangeData.SNARK_SCALAR_FIELD()` is only used here, should we just declare a constant var locally?

I think it makes sense to do it this way because it may be used elsewhere in the future. Seems to be inline with how we use the other constant values there as well.

> Should we check pubKeyX/Y before account creation and password reset instead of at places where these values are used?

- Any account creation/account update will also do a deposit so the check will be done in these cases
- The deposit is the place where the data is sent to the circuit

So I think it makes sense to do it there to make sure the check happens in all cases (technically it's not even needed).